model/riscv_insts_zvknhab.sail

```diff
@@ -0,0 +1,256 @@
+}
+
+val ROTR : forall 'n 'm, 'm >= 0. (bits('n), bits('m), bits('m)) -> bits('n)
+function ROTR (x, n, SEW) = (x >> n) | (x << (SEW - n))
+
+val SHR : forall 'n 'm, 'm >= 0. (bits('n), bits('m)) -> bits('n)
+function SHR (x, n) = x >> n
+
+
+val sig0 : forall 'n, 'n >= 0. (bits('n), int) -> bits('n)
+function sig0 (x, SEW) = {
+  let sew_bits : bits('n) = to_bits('n, SEW);
+  match SEW {
+    32 => (ROTR(x, to_bits('n, 7), sew_bits) ^ ROTR(x, to_bits('n, 18), sew_bits) ^ SHR(x, to_bits('n, 3))),
+    64 => (ROTR(x, to_bits('n, 1), sew_bits) ^ ROTR(x, to_bits('n, 8),  sew_bits) ^ SHR(x, to_bits('n, 7))),
+  }
+}
+
+val sig1 : forall 'n, 'n >= 0. (bits('n), int) -> bits('n)
+function sig1 (x, SEW) = {
+  let sew_bits : bits('n) = to_bits('n, SEW);
+  match SEW {
+    32 => (ROTR(x, to_bits('n, 17), sew_bits) ^ ROTR(x, to_bits('n, 19), sew_bits) ^ SHR(x, to_bits('n, 10))),
+    64 => (ROTR(x, to_bits('n, 19), sew_bits) ^ ROTR(x, to_bits('n, 61), sew_bits) ^ SHR(x, to_bits('n, 6 ))),
+  }
+}
+
+val ch : forall 'n, 'n >= 0. (bits('n), bits('n), bits('n)) -> bits('n)
+function ch (x, y, z) = ((x & y) ^ (not_vec(x) & z))
+
+val maj : forall 'n, 'n >= 0. (bits('n), bits('n), bits('n)) -> bits('n)
+function maj (x, y, z) = ((x & y) ^ (x & z) ^ (y & z))
+
+val sum0 : forall 'n, 'n >= 0. (bits('n), int) -> bits('n)
+function sum0 (x, SEW) = {
+  let sew_bits : bits('n) = to_bits('n, SEW);
+  match SEW {
+    32 => (ROTR(x, to_bits('n, 2), sew_bits)  ^ ROTR(x, to_bits('n, 13), sew_bits) ^ ROTR(x, to_bits('n, 22), sew_bits)),
+    64 => (ROTR(x, to_bits('n, 28), sew_bits) ^ ROTR(x, to_bits('n, 34), sew_bits) ^ ROTR(x, to_bits('n, 39), sew_bits)),
+  }
+}
+
+val sum1 : forall 'n, 'n >= 0. (bits('n), int) -> bits('n)
+function sum1 (x, SEW) = {
+  let sew_bits : bits('n) = to_bits('n, SEW);
+  match SEW {
+    32 => (ROTR(x, to_bits('n, 6), sew_bits)  ^ ROTR(x, to_bits('n, 11), sew_bits) ^ ROTR(x, to_bits('n, 25), sew_bits)),
+    64 => (ROTR(x, to_bits('n, 14), sew_bits) ^ ROTR(x, to_bits('n, 18), sew_bits) ^ ROTR(x, to_bits('n, 41), sew_bits)),
+  }
+}
+
+/* VSHA2MS.VV */
+
+union clause ast = RISCV_VSHA2ms : (regidx, regidx, regidx)
+
+mapping clause encdec = RISCV_VSHA2ms(vs2, vs1, vd) if (haveRVV() & (haveZvknha() | haveZvknhb()))
+ <-> 0b1011011 @ vs2 @ vs1 @ 0b010 @ vd @ 0b1110111 if (haveRVV() & (haveZvknha() | haveZvknhb()))
+
+mapping clause assembly = RISCV_VSHA2ms(vs2, vs1, vd)
+ <-> "vsha2ms.vv" ^ spc() ^ vreg_name(vd) ^ sep() ^ vreg_name(vs1) ^ vreg_name(vs2)
+
+function clause execute (RISCV_VSHA2ms(vs2, vs1, vd)) = {
+  let SEW      = get_sew();
+  let LMUL_pow = get_lmul_pow();
+  let LMUL     = if LMUL_pow < 0 then 0 else LMUL_pow;
+  let VLEN     = int_power(2, get_vlen_pow());
+  let num_elem = get_num_elem(LMUL_pow, SEW);
+
+  if (zvk_check_elements(VLEN, num_elem, LMUL, SEW) == false)
+  then {
+    handle_illegal();
+    RETIRE_FAIL
+  } else {
+    let 'n = num_elem;
+    let 'm = SEW;
```

Comment:
why no assert of `SEW` this time (only `32` and `64` should be supported depending on Zvknhb vs Zvknha)